PXC-2079 : Crash during rollback of a stored procedure called in an e…

kennt-percona opened the pull request:
https://jenkins.percona.com/job/pxc56.test.mtr/508/

kbauskar commented:
https://jenkins.percona.com/job/pxc56.test.mtr/508/

kbauskar commented on sql/event_scheduler.cc:
```diff
@@ -323,6 +324,34 @@ Event_worker_thread::run(THD *thd, Event_queue_element_for_exec *event)
+    We need to set up the proper thread state for the event worker thread
+    (so that WSREP will run/cleanup the thread correctly).
+
+    - wsrep_exc_state = LOCAL_STATE
```
s/wsrep_exc_state/wsrep_exec_state/gc

kennt-percona committed:
PXC-2079 : Crash during rollback of a stored procedure called in an event
PXC-2086 : BF aborted event does not update wsrep_local_bf_aborts variable

Issue:
PXC was not setting up the proper thread state for the event worker thread.
Thus, various WSREP-related actions were not being performed correctly
(such as rollback and the bf abort counter).

Solution:
Setup the event worker thread state to function more like a normal
local thread (not a replication thread).
Also, call the proper rollback thread if the thread is aborted.
